### ISSUE TITLE
Segmentation fault when using `INSERT OR REPLACE INTO`

### ISSUE THREAD

**Issue by hermanschaaf** [marcboeker/go-duckdb#80]
Thank you for building this library, it's been great so far. 

I ran into an error with `INSERT OR REPLACE INTO` queries, as the following fairly minimal reproduction illustrates:

```
package main

import (
	"database/sql"

	_ "github.com/marcboeker/go-duckdb"
)

func main() {
	db, err := sql.Open("duckdb", "")
	if err != nil {
		panic(err)
	}
	_, err = db.Exec("CREATE TABLE foo (id int PRIMARY KEY, b text);")
	if err != nil {
		panic(err)
	}
	_, err = db.Exec("INSERT OR REPLACE INTO foo VALUES (?, ?);", 1, "b")
	if err != nil {
		panic(err)
	}
	err = db.Close()
	if err != nil {
		panic(err)
	}
}
```

This exits with a fatal error on the `INSERT OR REPLACE INTO` line (which I'm truncating slightly for brevity):

```
fatal error: unexpected signal during runtime execution
[signal SIGSEGV: segmentation violation code=0x2 addr=0x8 pc=0x1052ed7b0]

runtime stack:
runtime.throw({0x1060b1960?, 0xaaaaaaaaaaaaaaab?})
        /opt/homebrew/Cellar/go/1.20.1/libexec/src/runtime/panic.go:1047 +0x40 fp=0x16b0623c0 sp=0x16b062390 pc=0x104dd1c50
runtime.sigpanic()
        /opt/homebrew/Cellar/go/1.20.1/libexec/src/runtime/signal_unix.go:821 +0x240 fp=0x16b062400 sp=0x16b0623c0 pc=0x104de7bc0

goroutine 1 [syscall]:
runtime.cgocall(0x104e58908, 0x14000058778)
        /opt/homebrew/Cellar/go/1.20.1/libexec/src/runtime/cgocall.go:157 +0x50 fp=0x14000058740 sp=0x14000058700 pc=0x104da4310
github.com/marcboeker/go-duckdb._Cfunc_duckdb_pending_prepared(0x6000027d42e0, 0x140000b2050)
        _cgo_gotypes.go:1319 +0x34 fp=0x14000058770 sp=0x14000058740 pc=0x104e52754
github.com/marcboeker/go-duckdb.(*stmt).execute.func1(0x140000587f8?, 0x140000960a0?)
        /Users/hermanschaaf/go/pkg/mod/github.com/marcboeker/go-duckdb@v1.2.0/statement.go:198 +0x78 fp=0x140000587b0 sp=0x14000058770 pc=0x104e56c78
github.com/marcboeker/go-duckdb.(*stmt).execute(0x140000aa108, {0x1062c7468, 0x140000b0000}, {0x140000960a0?, 0x1062aa380?, 0x1062c6ab0?})
        /Users/hermanschaaf/go/pkg/mod/github.com/marcboeker/go-duckdb@v1.2.0/statement.go:198 +0x84 fp=0x14000058860 sp=0x140000587b0 pc=0x104e563a4
github.com/marcboeker/go-duckdb.(*stmt).ExecContext(0x14000090130?, {0x1062c7468?, 0x140000b0000?}, {0x140000960a0?, 0x14000058938?, 0x1?})
        /Users/hermanschaaf/go/pkg/mod/github.com/marcboeker/go-duckdb@v1.2.0/statement.go:158 +0x38 fp=0x140000588e0 sp=0x14000058860 pc=0x104e560c8
github.com/marcboeker/go-duckdb.(*conn).ExecContext(0x140000589c8?, {0x1062c7468, 0x140000b0000}, {0x1060b145b?, 0x140000960a0?}, {0x140000960a0, 0x2, 0x2})
        /Users/hermanschaaf/go/pkg/mod/github.com/marcboeker/go-duckdb@v1.2.0/connection.go:47 +0xd0 fp=0x14000058960 sp=0x140000588e0 pc=0x104e52cf0
database/sql.ctxDriverExec({0x1062c7468?, 0x140000b0000?}, {0x1062c7028?, 0x14000090130?}, {0x0?, 0x0?}, {0x1060b145b?, 0x1062bd868?}, {0x140000960a0?, 0x104da9fc0?, ...})
        /opt/homebrew/Cellar/go/1.20.1/libexec/src/database/sql/ctxutil.go:31 +0xac fp=0x140000589f0 sp=0x14000058960 pc=0x104e358dc
database/sql.(*DB).execDC.func2()
        /opt/homebrew/Cellar/go/1.20.1/libexec/src/database/sql/sql.go:1675 +0x12c fp=0x14000058ac0 sp=0x140000589f0 pc=0x104e3a64c
database/sql.withLock({0x1062c72b0, 0x140000c4090}, 0x14000058c78)
        /opt/homebrew/Cellar/go/1.20.1/libexec/src/database/sql/sql.go:3405 +0x7c fp=0x14000058b00 sp=0x14000058ac0 pc=0x104e3aa4c
database/sql.(*DB).execDC(0x140000acb60?, {0x1062c7468, 0x140000b0000}, 0x140000c4090, 0x10639b860?, {0x1060b145b, 0x29}, {0x14000058f48, 0x2, 0x2})
        /opt/homebrew/Cellar/go/1.20.1/libexec/src/database/sql/sql.go:1670 +0x19c fp=0x14000058d10 sp=0x14000058b00 pc=0x104e3a13c
database/sql.(*DB).exec(0x14000058da8?, {0x1062c7468, 0x140000b0000}, {0x1060b145b, 0x29}, {0x14000058f48, 0x2, 0x2}, 0xa0?)
        /opt/homebrew/Cellar/go/1.20.1/libexec/src/database/sql/sql.go:1655 +0xa0 fp=0x14000058d90 sp=0x14000058d10 pc=0x104e39f40
database/sql.(*DB).ExecContext.func1(0x38?)
        /opt/homebrew/Cellar/go/1.20.1/libexec/src/database/sql/sql.go:1634 +0x54 fp=0x14000058e00 sp=0x14000058d90 pc=0x104e39e14
database/sql.(*DB).retry(0x14000058ee8?, 0x14000058e90)
        /opt/homebrew/Cellar/go/1.20.1/libexec/src/database/sql/sql.go:1538 +0x4c fp=0x14000058e50 sp=0x14000058e00 pc=0x104e39c2c
database/sql.(*DB).ExecContext(0x1060aabba?, {0x1062c7468?, 0x140000b0000?}, {0x1060b145b?, 0x1400004a728?}, {0x14000058f48?, 0x1400004a768?, 0x104dd41dc?})
        /opt/homebrew/Cellar/go/1.20.1/libexec/src/database/sql/sql.go:1633 +0x94 fp=0x14000058ef0 sp=0x14000058e50 pc=0x104e39d34
database/sql.(*DB).Exec(...)
        /opt/homebrew/Cellar/go/1.20.1/libexec/src/database/sql/sql.go:1647
main.main()
        /Users/hermanschaaf/code/cloudquery/cloudquery/plugins/destination/duckdb/bugs/test.go:18 +0xe4 fp=0x14000058f70 sp=0x14000058ef0 pc=0x104e57194
runtime.main()
```

If I change the query to a normal `INSERT INTO`, it works fine, so it seems specific to `INSERT OR REPLACE INTO`. I'm using `github.com/marcboeker/go-duckdb v1.2.0`.

I haven't dug into the internals of this library yet, so any pointers would be appreciated.

**Comment by marcboeker** [comment]
Hi @hermanschaaf 

thanks for reporting this issue. I've tried to reproduce it, but had no success. I extended your code to query for the inserted row and it works successfully. 

Are you using our precompiled Darwin arm64 libs or the ones provided by DuckDB? Or are you compiling it yourself?

Could you please upgrade to go-duckdb v1.2.1 which is based on DuckDB 0.7.1, a bugfix release.

Thanks!

**Comment by hermanschaaf** [comment]
Hi @marcboeker, thanks, that's great, it seems like v1.2.1 fixed the issue! I can still reproduce it on v1.2.0, but if I upgrade to v1.2.1 it works.

I wasn't doing anything special in terms of precompiled libs: just using the one embedded with the driver.

**Comment by marcboeker** [comment]
Great to hear. Feel free to reopen this issue, if you encounter this again.
